draft-irtf-cfrg-opaque.md

```diff
@@ -428,39 +429,52 @@ with the following API:
   `params` to strengthen the input `msg` against offline dictionary attacks.
   This function also needs to satisfy collision resistance.
 
-## Authenticated Key Exchange (AKE) Protocol {#deps-ake}
+## Key Recovery Method {#deps-keyrec}
 
-OPAQUE additionally depends on an Authenticated Key Exchange (AKE) protocol.
-This specification defines one particular AKE based on 3DH; see {{ake-protocol}}.
-3DH assumes a prime-order group as described in {{I-D.irtf-cfrg-voprf, Section 2.1}}.
-We let `Npk` and `Nsk` denote the size of public and private keys, respectively,
-used in the AKE. The AKE protocol must provide the following functions:
+OPAQUE relies on a key recovery mechanism so that a client can engage in the AKE.
+The key recovery mechanism MUST enable deterministic recovery of authentication material
+compatible with the AKE given the OPRF output and additional data.
+This additional data can be encapsulated in a specific `Envelope` structure, saved on
+the server during registration, and provided to the client during login.
```

Comment:
```suggestion
OPAQUE relies on a key recovery mechanism for storing and recovering authentication
material on the server. The key recovery storage mechanism takes as input a private 
seed and outputs an envelope. The retrieval process takes as input a private seed
and envelope and outputs authentication material. The signatures for these 
functionalities are below:

- Store(private_seed): build and return an `Envelope` structure
- Recover(private_seed, envelope): recover and return the authentication material
for the AKE from the Envelope.

The key recovery mechanism MUST be compatible with the chosen AKE.
```

Simplifying?

Reply:
Yeah, much better, thanks!
I suggest a little tweak though, to avoid any confusion on where the recovery takes place:

```suggestion
OPAQUE relies on a key recovery mechanism for storing authentication
material on the server and recovering it on the client. The key recovery storage
mechanism takes as input a private seed and outputs an envelope. The retrieval
process takes as input a private seed and envelope and outputs authentication
material. The signatures for these functionalities are below:

- Store(private_seed): build and return an `Envelope` structure
- Recover(private_seed, envelope): recover and return the authentication material
for the AKE from the Envelope.

The key recovery mechanism MUST be compatible with the chosen AKE.
```

 Also, before _"The key recovery storage mechanism takes as input a private seed and outputs an envelope."_ we don't define earlier what an envelope is. Do you think we should or leave it like that?

Reply:
I re-added `and the client's public key.` to the `Store()` signature, as it's necessary